include/crocoddyl/core/solver-base.hpp

```diff
@@ -76,8 +76,8 @@ class SolverAbstract {
    * @return A boolean that describes if convergence was reached.
    */
   virtual bool solve(const std::vector<Eigen::VectorXd>& init_xs = DEFAULT_VECTOR,
-                     const std::vector<Eigen::VectorXd>& init_us = DEFAULT_VECTOR, const std::size_t& maxiter = 100,
-                     const bool& is_feasible = false, const double& reg_init = 1e-9) = 0;
+                     const std::vector<Eigen::VectorXd>& init_us = DEFAULT_VECTOR, const std::size_t maxiter = 100,
+                     const bool is_feasible = false, const double reg_init = 1e-9) = 0;
```

Comment:
As explained before, I wanted to keep the same initial regularization value. But it is also important to define robustness rather than performance. I also explained above that this design choice was done by Nicolas, and I am inclined to this strategy as well.

As in other comments, I will close this discussion as I consider done. Do not hesitate to re-open it.

Reply:
We should set reg_init to reg_min here as well

Reply:
> We should set reg_init to reg_min here as well

I don't understand this comment. The value of reg_min is `1e-9` (as it was originally). We cannot neither write a code like this:
```cpp
                     const std::vector<Eigen::VectorXd>& init_us = DEFAULT_VECTOR, const std::size_t maxiter = 100,
                     const bool is_feasible = false, const double reg_init = reg_min_) = 0;
```
as `reg_min_` is a non-static data member.

Reply:
Ah yes. you are right, it has to be set in the Derived classes.